ietfparse/algorithms.py

```diff
@@ -160,51 +183,64 @@ def extract_quality(obj):
     return matches[0].candidate, matches[0].pattern
 
 
-def rewrite_url(input_url, **kwargs):
+unspecified_str = '...'
+
+
+def rewrite_url(input_url: str,
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this function is 26. By comparison, 99% of the functions in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the function has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this function or break it into multiple functions. For example, consider extracting the code block on lines 304-355 into a separate function.

Reply:
This is part of the reason that I am planning to deprecate and then remove the URL related functionality.  The next PR will officially deprecate the URL rewriting with a suggestion to use aio-libs/yarl